### PR TITLE
Instance: VM live migration during intra-cluster instance moves on ceph

### PR DESCRIPTION
**Pull request by tomponline**
Related to: https://discuss.linuxcontainers.org/t/lxd-online-vm-live-migration-qemu-to-qemu/16635

This PR adds support for VM QEMU to QEMU live migration during intra-cluster instance moves on ceph.

Because ceph provides shared storage when used for intra-cluster instance moves, only the VM's state is transferred, and the storage is left alone.

There is no public API change so I have not included an API extension in this PR.

Tests:

- For backwards compatibility with to/from LXD 5.12 (before the connection establishment logic changed).
- Live QEMU to QEMU intra-cluster move on ceph.
- Live stateful stop/stateful start intra-cluster move in ZFS.
- Live stateful stop/stateful start remote->remote move on dir and ZFS.

### CODE REVIEW

**Comment by simondeziel** [comment]
@tomponline I can't review anything in there but I /think/ that `shared/version/api.go` would need a new `APIExtensions` for this feature.

**Comment by simondeziel** [comment]
I see, I thought this API extension list was also for discoverability of supported features.

**Comment by tomponline** [comment]
> 

Yeah the actual VM live migration feature will need an API extension for sure (but that is way off yet as I've got lots left to do here).
We will also need extensions to the migration protocol itself for systems to negotiate how to do stateful migration (the current way using a state file that is copied) or the new way using a state socket.

But the constants themselves in `shared/api` shouldn't need one.

**Comment by simondeziel** [comment]
Ah that's crystal clear now, thanks for the explanation. I guess my enthusiasm was a tad premature! Sorry for commenting on a draft PR, it's just that I noticed we sometimes fail to add the API extension when new features land.

File: lxd/migration/migrate.pb.go [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
-// 	protoc-gen-go v1.28.1
-// 	protoc        v3.21.12
+// 	protoc-gen-go v1.28.0
+// 	protoc        v3.20.1
```
Comment (monstermunchkin) [inline lxd/migration/migrate.pb.go:6]: What's the reason for this change? I don't think we should downgrade the protobuf version.

Comment (tomponline) [reply lxd/migration/migrate.pb.go:6]: Mmm yeah looks like my protoc  version is older. Another reason to dislike that protobuf dependency.

Comment (tomponline) [reply lxd/migration/migrate.pb.go:6]: I'll see if I can get a newer one on my system somehow

Comment (tomponline) [reply lxd/migration/migrate.pb.go]: @stgraber @monstermunchkin that is fixed now.

**Comment by tomponline** [comment]
jenkins: test this please

**Comment by tomponline** [comment]
jenkins: test this please
